metrics-operator/controllers/analysis/workersPool.go

```diff
@@ -0,0 +1,94 @@
+package analysis
+
+import (
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis"
+	metricstypes "github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/providers"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate moq -pkg fake -skip-ensure -out ./fake/analysispool_mock.go . IAnalysisPool
+type IAnalysisPool interface {
+	DispatchAndCollect(ctx context.Context) (map[string]metricsapi.ProviderResult, error)
+}
+
+type NewWorkersPoolFactory func(ctx context.Context, analysis *metricsapi.Analysis, definition *metricsapi.AnalysisDefinition, numWorkers int, c client.Client, log logr.Logger, namespace string) (context.Context, IAnalysisPool)
+
+func NewWorkersPool(ctx context.Context, analysis *metricsapi.Analysis, definition *metricsapi.AnalysisDefinition, numWorkers int, c client.Client, log logr.Logger, namespace string) (context.Context, IAnalysisPool) {
```

Comment:
it seems like we only need the list of objectives of the `definition` parameter here, so I would suggest to pass a `[]metricsapi.Objectives` instead of `*metricsapi.AnalysisDefinition` (this should also make the incorporation of https://github.com/keptn/lifecycle-toolkit/pull/1875/files#r1311338157 a bit easier)